runtime-modules/referendum/src/benchmarking.rs

```diff
@@ -0,0 +1,739 @@
+    T::create_option(account_option, member_option);
+
+    vote_for::<T, I>(name, voter_id, member_option, cycle_id, extra_stake)
+}
+
+fn move_to_block<T: Trait<I>, I: Instance>(
+    target_block: T::BlockNumber,
+    target_stage: ReferendumStage<T::BlockNumber, T::MemberId, T::VotePower>,
+) {
+    let mut current_block_number = System::<T>::block_number();
+
+    while current_block_number < target_block {
+        System::<T>::on_finalize(current_block_number);
+        Referendum::<T, I>::on_finalize(current_block_number);
+        current_block_number = System::<T>::block_number() + One::one();
+        System::<T>::set_block_number(current_block_number);
+        System::<T>::on_initialize(current_block_number);
+        Referendum::<T, I>::on_initialize(current_block_number);
+    }
+
+    assert_eq!(Stage::<T, I>::get(), target_stage, "Stage not reached");
+}
+
+fn get_byte(num: u32, byte_number: u8) -> u8 {
+    ((num & (0xff << (8 * byte_number))) >> 8 * byte_number) as u8
+}
+
+// Method to generate a distintic valid handle
+// for a membership. For each index.
+fn handle_from_id<T: Trait<I> + membership::Trait, I: Instance>(id: u32) -> Vec<u8> {
+    let mut handle = vec![];
+
+    for i in 0..4 {
+        handle.push(get_byte(id, i));
+    }
+
+    handle
+}
+
+fn member_funded_account<T: Trait<I> + membership::Trait, I: Instance>(
+    id: u32,
+) -> (T::AccountId, T::MemberId) {
+    let account_id = funded_account::<T, I>("account", id);
+
+    let handle = handle_from_id::<T, I>(id);
+
+    let member_id = Membership::<T>::members_created();
+
+    let params = membership::BuyMembershipParameters {
+        root_account: account_id.clone(),
+        controller_account: account_id.clone(),
+        name: None,
+        handle: Some(handle),
+        avatar_uri: None,
+        about: None,
+        referrer_id: None,
+    };
+
+    Membership::<T>::buy_membership(RawOrigin::Signed(account_id.clone()).into(), params).unwrap();
+
+    T::Currency::make_free_balance_be(&account_id, Balance::<T, I>::max_value());
+
+    Membership::<T>::add_staking_account_candidate(
+        RawOrigin::Signed(account_id.clone()).into(),
+        member_id,
+    )
+    .unwrap();
+
+    Membership::<T>::confirm_staking_account(
+        RawOrigin::Signed(account_id.clone()).into(),
+        member_id,
+        account_id.clone(),
+    )
+    .unwrap();
+
+    (account_id, member_id)
+}
+
+fn add_and_reveal_multiple_votes_and_add_extra_unrevealed_vote<
+    T: Trait<I>
+        + OptionCreator<<T as frame_system::Trait>::AccountId, <T as common::Trait>::MemberId>
+        + membership::Trait,
+    I: Instance,
+>(
+    target_winners: u32,
+    number_of_voters: u32,
+    extra_vote_option: u32,
+    extra_stake: Balance<T, I>,
+) -> (
+    Vec<OptionResult<T::MemberId, T::VotePower>>,
+    T::AccountId,
+    T::MemberId,
+    T::Hash,
+) {
+    start_voting_cycle::<T, I>(target_winners);
+
+    let cycle_id = 0;
```

Comment:
Sometimes you use cycle_id (=0) and later you use just 0. Make it consistent, please. In other places also.